base/compiler/ssair/passes.jl

```diff
@@ -720,6 +720,87 @@ function perform_lifting!(compact::IncrementalCompact,
     return stmt_val # N.B. should never happen
 end
 
+function lift_svec_ref!(compact::IncrementalCompact, idx::Int, stmt::Expr)
+    if length(stmt.args) != 4
+        return
+    end
+
+    vec = stmt.args[3]
+    val = stmt.args[4]
+    valT = argextype(val, compact)
+    (isa(valT, Const) && isa(valT.val, Int)) || return
+    valI = valT.val::Int
+    (1 <= valI) || return
+
+    if isa(vec, SimpleVector)
+        if valI <= length(val)
+            compact[idx] = vec[valI]
+        end
+        return
+    end
+
+    if isa(vec, SSAValue)
+        # TODO: We could do the whole lifing machinery here, but really all
+        # we want to do is clean this up when it got inserted by inlining,
+        # which always targets simple `svec` call or `_compute_sparams`,
+        # so this specialized lifting would be enough
+        def = compact[vec]
+        if is_known_call(def, Core.svec, compact)
+            nargs = length(def.args)
+            if valI <= nargs-1
+                compact[idx] = def.args[valI+1]
+            end
+            return
+        elseif is_known_call(def, Core._compute_sparams, compact)
+            m = argextype(def.args[2], compact)
+            isa(m, Const) || return
+            m = m.val
+            isa(m, Method) || return
+            # For now, just pattern match the benchmark case
+            # TODO: More general structural analysis of the intersection
+            length(def.args) == 3 || return
+            sig = m.sig
+            isa(sig, UnionAll) || return
+            tvar = sig.var
+            sig = sig.body
+            isa(sig, DataType) || return
+            sig.name === Tuple.name
```

Comment:
This does nothing.

Reply:
I take it we want `sig.name === Tuple.name || return`?